clippy_lints/src/methods/manual_try_fold.rs

```diff
@@ -0,0 +1,55 @@
+use clippy_utils::{
+    diagnostics::span_lint_and_sugg,
+    is_from_proc_macro,
+    msrvs::{Msrv, ITERATOR_TRY_FOLD},
+    source::snippet_opt,
+    ty::implements_trait,
+};
+use rustc_errors::Applicability;
+use rustc_hir::{
+    def::{DefKind, Res},
+    Expr, ExprKind,
+};
+use rustc_lint::{LateContext, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_span::Span;
+
+use super::MANUAL_TRY_FOLD;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &Expr<'tcx>,
+    init: &Expr<'_>,
+    acc: &Expr<'_>,
+    fold_span: Span,
+    msrv: &Msrv,
+) {
+    if !in_external_macro(cx.sess(), fold_span)
+        && msrv.meets(ITERATOR_TRY_FOLD)
```

Comment:
> Can you make sure that it's also added to the documentation in the conf.rs file, to make it link correctly in the docs?

[Here](https://github.com/rust-lang/rust-clippy/blob/10ce1a6d5e7c36cf9b3cae12baf8344726e6de93/clippy_lints/src/utils/conf.rs#L297)

As a a side note, having an automated way (like `update_lints`) to check this could be reeeally useful and I may or may not start doing it right now.

Reply:
I'm uncertain if `update_lints` has enough information for this analysis/update. This is one of the reasons, why we considered moving the configuration declaration/linking to the lint declaration. We sadly haven't found a nice way to handle it uniformly for all lints.